gh-actions/C/format-source.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Configure here all the directories where are source files
+SOURCE_FILES="src/*.[ch] tests/*.[ch]"
```

Comment:
These should be configurables as an action `input`, ideally you can just rely on `find \( -name '*.c' -or -name '*.h' \) -exec sh -c 'clang-format --dry-run --Werror "$1" || kill $PPID' -- {} \;`

Reply:
Mmm... good point... Give me a moment.

Reply:
The point is that the same script must be able to be run from the shell: you want to also run it from pre-commit hook to avoid unnecessary CI tests, and manually to format the code. And the idea is to use exactly the same script in those three cases to ensure consistency (in fact, this idea came when I formatted a commit with what I believed was a correct clang-format configuration, only to discover after the upload that it wasn't). That means that we can't use action inputs. About using find... it can be a good idea. I'll give it a spin.

Reply:
Ok, I gave it a spin, and has the problem that it also formats any .c and .h file inside the `build` folder too (the ones generated by meson; for example the code generated by `gdbus-codegen`). That's not really a problem, just an annoyance. Anyway, here is the change. It required some tricks to work, anyway... Give you opinion, please.